tests/test-sessions/test-mongodb-sessions/src/test/java/org/eclipse/jetty/nosql/mongodb/PurgeValidSessionTest.java

```diff
@@ -123,6 +123,75 @@ public void testPurgeValidSession() throws Exception
     }
 
 
+    @Test
+    public void testPurgeValidSessionWithPurgeLimitSet() throws Exception
```

Comment:
This should be fixed now. Added handling to deal with existing sessions in the store that aren't going to be invalidated.